### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): api for morphism properties local at the source and target

### PR DESCRIPTION
**Pull request by chrisflav**
A morphism property, that is both local at the source and the target and additionally is stable under post-composition with open immersions, can be checked locally around points of the source. This is what the [stacks](https://stacks.math.columbia.edu/tag/04QW) project calls "local on the source-and-target".

---

- [x] depends on: #17264

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Morphisms/Basic.lean [hunk 1]
```diff
@@ -270,6 +275,32 @@ lemma isLocalAtTarget [P.IsMultiplicative]
     · exact hP _ _
     · exact fun H ↦ P.comp_mem _ _ H (of_isOpenImmersion _)
 
+section IsLocalAtSourceAndTarget
+
+/-- If `P` is local at the source and the target, then restriction on both source and target
+preserves `P`. -/
+lemma resLE_of_isLocalAtTarget [IsLocalAtTarget P] {U : Y.Opens} {V : X.Opens} (e : V ≤ f ⁻¹ᵁ U)
```
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/Morphisms/Basic.lean:282]:
```suggestion
lemma resLE [IsLocalAtTarget P] {U : Y.Opens} {V : X.Opens} (e : V ≤ f ⁻¹ᵁ U)
```
I think this is enough?

File: Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean [hunk 2]
```diff
@@ -192,6 +192,9 @@ lemma HasAffineProperty : HasAffineProperty P (sourceAffineLocally Q) where
     (isLocal_ringHomProperty P).ofLocalizationSpan
   eq_targetAffineLocally' := eq_affineLocally P
 
+/- This is only `inferInstance` because of the `@[local instance]` on `HasAffineProperty` above. -/
+instance : IsLocalAtTarget P := inferInstance
```
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean:196]:
```suggestion
instance (priority := 900) : IsLocalAtTarget P := inferInstance
```

**Review by erdOne — APPROVED** [review]
LGTM modulo two minor comments

**Comment by jcommelin** [comment]
maintainer merge

**Pull request closed without merging**